### PR TITLE
Automatically choose correct socketcan interface

### PR DESCRIPTION
**Pull request by hardbyte**
Note I also moved `can.interfaces.interface` to `can.interface` as it seems to make more sense there.

Fix for #120

### CODE REVIEW

File: can/interface.py [hunk 1]
```diff
@@ -27,21 +23,12 @@ def __new__(cls, other, channel=None, *args, **kwargs):
             can.rc['interface'] = kwargs['bustype']
```
Comment (christiansandberg) [inline can/interface.py]: I've been thinking about this. Wouldn't be more logical to never change can.rc if a custom bustype has been given? One might think that the default interface and channel should always be used when nothing else has been specified when initiating the Bus, but now it depends on the last created bus.

Comment (hardbyte) [inline can/interface.py]: Yes - I agree in principal. A read only global config variable would be an improvement. I'll take a look and merge in the 'neovi' + io changes.

Comment (christiansandberg) [inline can/interface.py]: It doesn't have to be read only (one might still want to change it programmatically), just that creating a bus different from the default should not modify the default.

Comment (hardbyte) [inline can/interface.py]: Yup I meant the library should treat it as read only - the user may of course change at will.

**Comment by hardbyte** [comment]
I'd appreciate it if you could cast your eyes over those changes too

**Comment by christiansandberg** [comment]
I think it looks good!

**Pull request merged**